src/hotspot/share/prims/jvmtiExport.cpp

```diff
@@ -1094,6 +1093,9 @@ bool JvmtiExport::post_class_file_load_hook(Symbol* h_name,
   if (JavaThread::current()->is_in_tmp_VTMS_transition()) {
     return false; // skip CFLH events in tmp VTMS transition
   }
+  if (JavaThread::current()->is_in_any_VTMS_transition()) {
+    return false; // no events should be posted if thread is in any VTMS transition
+  }
```

Comment:
Sorry, I was not clear the 3 lines above 1093-1095 had to be replaced with new lines 1096-1098.
The check for `is_in_any_VTMS_transition()` includes the checks `is_in_tmp_VTMS_transition()`.

Reply:
Change it.